addons/website_sale_stock/static/src/js/website_sale_stock.js

```diff
@@ -0,0 +1,6 @@
+$(document).ready(function () {
+    //show warning outofstock, when click on add to cart from shop ,trigger on change of quantity input
+    if(window.location.pathname === '/shop/cart' && $('.oe_cart .show_outofstock_warning').length) {
+        $('.oe_cart input.js_quantity').trigger('change');
+    }
+});
```

Comment:
@vme-odoo  : when we click on "Add to Cart"  button from product/general shop view , we are not able to trigger "js_quantity" (quantity input box), so when we click on "Add to Cart" and redirect to "/shop/cart" , we need to forcefully trigger a quantity input box to show alert , i think this is better way to do it.